libvast/src/system/indexer.cpp

```diff
@@ -38,32 +39,50 @@ indexer_state::~indexer_state() {
   col.~column_index();
 }
 
-caf::error indexer_state::init(caf::event_based_actor* self, path filename,
-                               type column_type, caf::settings index_opts,
-                               caf::actor index, uuid partition_id,
-                               atomic_measurement* m) {
+caf::error
+indexer_state::init(caf::event_based_actor* self, path filename,
+                    type column_type, caf::settings index_opts,
+                    caf::actor index, uuid partition_id, std::string fqn) {
   this->index = std::move(index);
   this->partition_id = partition_id;
-  this->measurement = m;
+  this->fqn = fqn;
+  this->self = self;
+  this->streaming_done = false;
```

Comment:
I think we should only enable this when the index is currently under construction.
```suggestion
```